pkg/compiler/check.go

```diff
@@ -373,7 +373,10 @@ func (comp *compiler) checkLenTarget(t *ast.Type, name, target string, fields []
 }
 
 func CollectUnused(desc *ast.Description, target *targets.Target) []ast.Node {
-	comp := createCompiler(desc, target, nil)
+	eh := func(pos ast.Pos, msg string) {
```

Comment:
It seems that we can make sys/fuchsia/fidlgen properly handle errors with just a little bit more effort.

First, we need to move:
```
	if eh == nil {
		eh = ast.LoggingHandler
	}
```
from Compile function to createCompiler.

Second, return an error from CollectUnused if typecheck finds any errors. See how Compile function detects typecheck errors.

Finally, check the error in fidlgen.
Then we will have fidlgen just properly print all errors.

Reply:
Thanks for the suggestions! I made the changes, and I also adding error checking after calling `CollectUnused`. Right now it doesn't add any errors, but maybe in the future it will.

What would be the best way to add a test for this new behavior?

Reply:
What exactly aspect you want to test?
We have tests in pkg/compiner/compier_test.go. You could parse some simple description with an error and check that it fails with an error. Is it what you want to test?

Reply:
I added a test for the basic functionality of `CollectUnused` plus the error checking. PTAL